GNU uname aka. guname(1) -o, --operating-system is uname(1) -s.
Ignore whitespace (some diff'd output has trailing ^M).
Remove bash-ism.

Index: tests/testrun.sh
--- tests/testrun.sh.orig
+++ tests/testrun.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-OS_ARCH=`uname -o 2>/dev/null || echo ""`
+OS_ARCH=`uname -s 2>/dev/null || echo ""`
 
 #
 # Get command line params
@@ -413,11 +413,7 @@ if [ -n "$PERF_TEST" ] ; then
     xmlsec_params="$xmlsec_params --repeat $PERF_TEST"
 fi
 
-if test "z$OS_ARCH" = "zCygwin" || test "z$OS_ARCH" = "zMsys" ; then
-    diff_param=-uw
-else
-    diff_param=-u
-fi
+diff_param=-ub
 
 
 #
@@ -520,7 +516,7 @@ extra_message=""
 rm -rf $tmpfile $tmpfile.2 $tmpfile.3
 
 # run tests
-source "$testfile"
+. "$testfile"
 
 # calculate success
 percent_success=0
